guides/common/modules/proc_creating-a-local-source-for-a-custom-file-type-repository.adoc

```diff
@@ -0,0 +1,71 @@
+[id="Creating_a_Local_Source_for_a_Custom_File_Type_Repository_{context}"]
+= Creating a Local Source for a {customfiletypetitle} Repository
+
+You can create a {customfiletype} repository source, from a directory of files, on the base system where {Project} is installed using *Pulp Manifest*.
+You can then synchronize the files into {ProjectServer}.
+When you add files to a file type repository, you can work with the files as with any other repository.
+
+Use this procedure to configure a repository in a directory on the base system where {Project} is installed.
+To create a file type repository in a directory on a remote server, see xref:Creating_a_Remote_Source_for_a_Custom_File_Type_Repository_{context}[].
+
+.Procedure
+ifdef::satellite[]
+. Ensure the Utils repository is enabled.
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# subscription-manager repos --enable={RepoRHEL8BaseOS} \
+--enable={RepoRHEL8AppStream} \
+--enable={RepoRHEL8ServerSatelliteUtils}
+----
+endif::[]
+. Install the Pulp Manifest package:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# {package-install-project} python3-pulp_manifest
+----
+ifdef::satellite[]
++
+Note that this command stops the {Project} service and re-runs `{foreman-installer}`.
+Alternatively, to prevent downtime caused by stopping the service, you can use the following:
++
+[options="nowrap" subs="+quotes,attributes"]
+----
+# {foreman-maintain} packages unlock
+# {package-install} python3-pulp_manifest
+# {foreman-maintain} packages lock
+----
+endif::[]
+. Create a directory that you want to use as the file type repository in the HTTP server's public folder:
```

Comment:
Does that imply that you'd need a specific path in line 44?

IMHO This is true because of `pulpcore_additional_import_paths` which means Foreman can only import (by default) from  the `/var/lib/pulp/imports/` directory if I'm not mistaken.

You could add your path, e.g. `/var/my_repo` to  `/etc/foreman-installer/scenarios.d/katello-answers.yaml` and then run `foreman-installer`.

I am sure there is a more elegant solution to this.

Reply:
Actually, I think that the path will be the same here as with the "remote source" option.
In both cases, the repository is synced over HTTP/HTTPS because I "import" these repositories via an Upstream URL.
So any other paths shouldn't be necessary.
Does that make sense?

Reply:
Okay, you're right! I got this
> If you do see some error saying not in allowed import paths, you might have to add it in /etc/pulp/settings.py in the ALLOWED_IMPORT_PATHS list..